src/classes/CW_BriefApprovalProcess.cls

```diff
@@ -0,0 +1,167 @@
+/**
+* File Name   :    CW_BriefApprovalProcess .cls
+* Description :    B-106804 As a Marketing Ops I want the ability to submit a MarCom Brief
+*
+* Modification Log
+* ============================================================================
+* Ver Date       Author         Modification
+* --- ---------- -------------- --------------------------
+* 0.1 24/04/2014 Cristina P.	Intial
+*
+*/
+public with sharing class CW_BriefApprovalProcess 
+{
+	public Set<String> 	briefFieldSet;
+	public String 		errorMessage;
+	public Boolean briefRequiresApproval;
+	public Boolean submitBriefForApproval;
+	
+	//check if required fields are populated
+	public Boolean checkBriefForApproval(CW_Campaign__c camp)
+ 	{
+		//loop though required fields from brief fieldset
+ 		for (CW_Campaign__c curr : getBriefRequiredFields(camp.Id))
+		{
+			for (String s : briefFieldSet)
+			{
+				if (curr.get(s)!=null && curr.get(s)!=0)
+					briefRequiresApproval = true;
+				else
+				{
+					briefRequiresApproval = false;
+					break;
+				}
+			}
+		}
+		//brand, label, language, country are required fields from brief approval
+		//check if there is a campaign site associated to the campiagn
+		List<CW_Campaign_Sites__c> campSites = [SELECT Name FROM CW_Campaign_Sites__c WHERE Campaign__c =: camp.Id];						
+ 		//fields required are all populated an there is campSite for the Brief
+ 		if (briefRequiresApproval && !campSites.isEmpty())
+ 		{
+ 			if (!camp.Urgent_Campaign__c)
+ 			{
+ 				// brief is linked to a roadmap
+	 			if (camp.Roadmap__c!=null)
+	 			{
+	 				//roadmap MUST be approved so Brief can be submitted
+		 			String RoadmapStatus = [SELECT Approval_Status__c 
+		 									FROM CW_Roadmap__c 
+		 									WHERE Id =: camp.Roadmap__c].Approval_Status__c;
+	 				
+	 				//roadmap APPROVED so brief can be submitted for approval
+			 		if (RoadmapStatus == ConstantUtil.APPROVAL_PROCESS_APPROVED)
```

Comment:
Nesting too many `IF` statements affects readability and makes maintenance more difficult. I suggest reviewing the design to simplify the flow of this code block.